pkg/resources/management.cattle.io/v3/project/mutator_test.go

```diff
@@ -0,0 +1,96 @@
+package project
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/generic/fake"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	expectedIndexerName = "webhook.cattle.io/creator-role-template-index"
+	expectedIndexKey    = "creatorDefaultUnlocked"
+)
+
+func TestAdmit(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name      string
+		operation admissionv1.Operation
+		project   *v3.Project
+		wantPatch []map[string]interface{}
+		wantErr   bool
+	}{
+		{
+			name:      "delete operation is invalid",
+			operation: admissionv1.Delete,
+			wantErr:   true,
+		},
+		{
+			name:      "update operation is invalid",
+			operation: admissionv1.Update,
+			wantErr:   true,
+		},
+		{
+			name:      "connect operation is invalid",
+			operation: admissionv1.Connect,
+			wantErr:   true,
+		},
```

Comment:
Right now I think these tests are producing an error because the mutator can't decode a project for them, not because it recognizes an invalid operation. This leaves the invalid operation line uncovered in the coverage report. I was able to fix this by adding the project in the created case to the update/connect case (the delete case requires that you set the old object not the current object, so you'll need to move the framework around a bit for that one).

Reply:
Fixed